script.js

```diff
@@ -1,4 +1,811 @@
+    // Inner loop for each of the 13 ranks
+    for (let rankIndex = 1; rankIndex <= 13; rankIndex++) {
+      // Card names being the card ranks except for face cards
+      let cardName = rankIndex;
+      if (rankIndex === 1) {
+        cardName = "Ace";
+      } else if (rankIndex === 11) {
+        cardName = "Jack";
+      } else if (rankIndex === 12) {
+        cardName = "Queen";
+      } else if (rankIndex === 13) {
+        cardName = "King";
+      }
+      // Create a single card object
+      const card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankIndex,
+      };
+      // If the card is jack, queen, king, or ace, update the card values in accordance to blackjack rules
+      // Ace is 11 by default and another function will account for ace being counted as 1
+      if (card.rank === 11 || card.rank === 12 || card.rank === 13) {
+        card.rank = 10;
+      } else if (card.rank === 1) {
+        card.rank = 11;
+      }
+      // Push card object to deck array
+      deck.push(card);
+    }
+  }
+  return deck;
+};
+
+// Functions to make a shuffled blackjack deck
+const getRandomNumber = function (max) {
+  return Math.floor(Math.random() * max);
+};
+const shuffleDeck = function (deck) {
+  // Loop through the entire deck
+  for (let currentIndex = 0; currentIndex < deck.length; currentIndex++) {
+    // Get the current card
+    let currentCard = deck[currentIndex];
+    // Get a random index number
+    let randomIndex = getRandomNumber(deck.length);
+    // Get a random card
+    let randomCard = deck[randomIndex];
+    // Swap the random card with the current card
+    deck[currentIndex] = randomCard;
+    deck[randomIndex] = currentCard;
+  }
+  return deck;
+};
+
+// --- The term 'user' is explicitly used to refer to either player or pc ---
+
+// Function to draw multiple random cards
+const drawCards = function (cardCount, userCards, deck) {
+  for (let i = 0; i < cardCount; i++) {
+    // Remove card object from the deck and push it to another array of player or pc's cards
+    userCards.push(deck.pop());
+  }
+  return userCards;
+};
+
+// Function to calculate each user's scores, i.e. sum of all their card values
+const calculateScores = function (userCards) {
+  // Initialize sum to 0
+  let sum = 0;
+  // Loop through the user card values and increment the sum by each of the card values
+  for (let i = 0; i < userCards.length; i++) {
+    sum += userCards[i].rank;
+  }
+
+  // Loop through the user cards to check if there is any ace card and if the sum from the previous loop is > 21, if so, reduce sum by 10 for each ace until the sum is <= 21. Examples are available the bottom of this file.
+  // Put it simply, reduce the sum by 10 only if the ace card causes the user to be busted. Ace is counted as 1 instead of 11 in that case
+  for (let j = 0; j < userCards.length; j++) {
+    if (userCards[j].rank === 11 && sum > 21) {
+      sum -= 10;
+    }
+  }
```

Comment:
can consider having this in its own function called `handleAce` or something like that, then you also won't have to write so much comment